### ISSUE TITLE
Regularization Loss is getting added twice if model is encapsulated in another model

### ISSUE THREAD

**Issue by ksachdeva** [tensorflow/tensorflow#43995]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Pre-installed TF on Colab
- TensorFlow version (use command below): 2.3
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

```
import tensorflow as tf
tfk = tf.keras
tfkl = tf.keras.layers

def test_model(input_shape):
  input_x = tfkl.Input(shape=input_shape)
  h = tfkl.Flatten()(input_x)
  h = tfkl.Dense(10, activation='relu', activity_regularizer=tf.keras.regularizers.l1())(h)

  return tf.keras.models.Model(inputs=input_x, outputs=h)

def another_model(input_shape, tm):
  input_x = tfkl.Input(shape=input_shape)
  m = tm(input_x)
  return tf.keras.models.Model(inputs=input_x, outputs=m)

tm = test_model(input_shape)
print(another_model(input_shape, tm).losses)

```
Above prints

```
[<tf.Tensor 'dense_27/ActivityRegularizer/truediv:0' shape=() dtype=float32>,
 <tf.Tensor 'functional_20/dense_27/ActivityRegularizer/truediv:0' shape=() dtype=float32>]
```

This pattern of creating simple models (modules) and then assembling them as part of the big model is widely used and as shown in the above code snippets and its outcome this ends up adding the regularization loss twice.

I noticed it while adding KLDivergenceRegularization from tensorflow_probability and then reproduced it using a simple example as shown above.

Please suggest if there is a workaround as I said earlier I have many models that I assemble.

Regards & thanks
Kapil

**Comment by Saduf2019** [comment]
@ksachdeva 
I ran the code shared and face a different issue, please refer to [this gist](https://colab.research.google.com/gist/Saduf2019/165c745200ca1aa88b0d5545d07d40c7/untitled433.ipynb) and share all dependencies or share a colab gist with error reported.

**Comment by ksachdeva** [comment]
Hi @Saduf2019 

You could use any input_shape e.g.  input_shape = [28, 28, 1]

Nevertheless, please find the link to the colab notebook 

https://colab.research.google.com/drive/1oa5fdFEuM2d_k-OrL4Rif8EcV0EJjavR?usp=sharing

Regards
Kapil

**Comment by Saduf2019** [comment]
i am able to replicate the issue reported, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/39b6b45eac74c96b51ebee77ebcd45e5/untitled437.ipynb).

**Comment by ksachdeva** [comment]
Hi,

Is there any action on this?

IMHO, this is a very dangerous bug as the practice of using Loss layers, activity regularization, and model wrapping is quite common. 

Regards
Kapil

**Comment by ksachdeva** [comment]
pinging again! 

@Saduf2019 since you were able to reproduce this is there a way to get a resolution on this.

**Comment by sachinprasadhs** [comment]
Was able to reproduce your issue in Tensorflow 2.5, please find the gist [here](https://colab.research.google.com/gist/sachinprasadhs/2a386bd801d28ce02473b187e8cb7ba3/43995.ipynb). Thanks!

**Comment by ksachdeva** [comment]
@sachinprasadhs seems like it is not a priority for the TensorFlow team. It is a very serious bug in my opinion. I have been a very big advocate of TensorFlow & Keras in my organization but this experience has left me questioning if it is indeed time to switch to PyTorch. Very disappointed!

**Comment by jvishnuvardhan** [comment]
@deeb02 Can you please take a look at this issue? Thanks!

**Comment by gowthamkpr** [comment]
@ksachdeva The development of Keas has been moved to keras-team/keras repo. Since this issue is already present [here](https://github.com/keras-team/keras/issues/15909), can we please go ahead and close this issue. Thank!!
